src/types.ts

```diff
@@ -5,17 +5,21 @@ import {
   FluentOptions,
   FluentVariable,
   LocaleId,
-  TranslationContext,
 } from "./deps.ts";
 
 export type LocaleNegotiator<C extends Context = Context> = (ctx: C) =>
   | LocaleId
   | undefined
   | PromiseLike<LocaleId | undefined>;
 
-export type TranslateFunction = (
+export type CustomTranslationContext<K extends string> = Record<
```

Comment:
Can we name it not something like Context? In relation with grammY it is confusing for users to have two contexts which are different things.

Fluent itself seems to call them ["placeables"](https://projectfluent.org/fluent/guide/placeables.html).

Reply:
That's taken from https://deno.land/x/better_fluent@v1.0.0/mod.ts?s=TranslationContext

Reply:
Also "Custom" is a bit… there only is one type for that, so its not a "custom" kind of it, its the default one.

Reply:
> That's taken from https://deno.land/x/better_fluent@v1.0.0/mod.ts?s=TranslationContext

Yeah, I know, but we dont need to do something another library does especially when the project itself doesnt seem to use the word "context" at all :thinking: 
<https://duckduckgo.com/?t=ffab&q=site%3Aprojectfluent.org+context&ia=web>

Reply:
Placeables seem to be in a translation, variables are filled into these placeables. So placeable would be wrong too. But naming them variables might not be that helpful.

Reply:
Dont get me wrong, I see where the name is coming from. But I think we can do better.

Reply:
I do have to agree that the terms custom and context and usually combined when referring to something that is very different …

Reply:
How about just naming it `TranslationVariables`?

Results basically in this type which seems easy to understand what it does and what its for:

```typescript
export type TranslationVariables<K extends string> = Record<K, FluentVariable>;
```

Reply:
I like that name! @Loskir?

Reply:
Me too

Reply:
Hmm, what's the point of Readonly?

Reply:
@dcdunkan what do you think? it would be cool to add these changes to better_fluent itself. However, it's probably breaking

Reply:
Readonly helps to ensure that the function does not have side effects on the input and does not change the input variable. (But its still only TypeScript, JavaScript has no types nor the readonly stuff and its not recursive. Not as good as I would like it to be but better than nothing)

Reply:
> @dcdunkan what do you think? it would be cool to add these changes to better_fluent itself. However, it's probably breaking

I agree with it. But I'm away from my PC and I won't be able to work on it for a few months. I'd appreciate if someone could work on this and create a PR to better_fluent. I can review and merge the changes.